spec/config.spec.ts

```diff
@@ -24,104 +24,127 @@ import * as mockRequire from 'mock-require';
 import { expect } from 'chai';
 import { config, firebaseConfig } from '../src/config';
 
-describe('config()', () => {
+const NODE8_NODE10_PATH = '/srv/.runtimeconfig.json';
+const NODE6_PATH = '/user_code/.runtimeconfig.json';
+
+describe.only('config()', () => {
   afterEach(() => {
     mockRequire.stopAll();
     delete config.singleton;
     delete process.env.FIREBASE_CONFIG;
-    delete process.env.FIREBASE_PROJECT;
-    delete process.env.CLOUD_RUNTIME_CONFIG;
+    delete process.env.PWD;
   });
 
-  it('loads config values from .runtimeconfig.json', () => {
-    mockRequire('../../../.runtimeconfig.json', { foo: 'bar', firebase: {} });
-    let loaded = config();
-    expect(loaded).to.not.have.property('firebase');
-    expect(loaded).to.have.property('foo', 'bar');
+  it('loads Firebase config from FIREBASE_CONFIG env variable', () => {
+    process.env.FIREBASE_CONFIG = JSON.stringify({
+      databaseURL: 'foo@firebaseio.com',
+    });
+
+    expect(firebaseConfig()).to.have.property(
+      'databaseURL',
+      'foo@firebaseio.com'
+    );
   });
 
   it('does not provide firebase config if .runtimeconfig.json not invalid', () => {
-    mockRequire('../../../.runtimeconfig.json', 'does-not-exist');
+    mockRequire(NODE8_NODE10_PATH, 'does-not-exist');
+
     expect(firebaseConfig()).to.be.null;
   });
 
-  it('does not provide firebase config if .ruuntimeconfig.json has no firebase property', () => {
-    mockRequire('../../../.runtimeconfig.json', {});
+  it('does not provide firebase config if .runtimeconfig.json has no firebase property', () => {
+    mockRequire(NODE8_NODE10_PATH, {});
+
     expect(firebaseConfig()).to.be.null;
   });
 
-  it('loads Firebase configs from FIREBASE_PROJECT env variable', () => {
-    process.env.FIREBASE_PROJECT = JSON.stringify({
-      databaseURL: 'foo@firebaseio.com',
+  it('loads firebase config from .runtimeconfig.json properly for Node 6 functions', () => {
+    process.env.PWD = '/user_code';
+    mockRequire(NODE6_PATH, {
+      foo: 'bar',
+      firebase: {
+        projectId: 'chenky-test',
+        databaseURL: 'https://chenky-test.firebaseio.com',
+        storageBucket: 'chenky-test.appspot.com',
+        cloudResourceLocation: 'us-central',
+      },
     });
-    expect(firebaseConfig()).to.have.property(
+    let loaded = firebaseConfig();
```

Comment:
Could you add a new line between setting up mocks and calling the method under test, in each test, to split them into arrange, act, assert style? bkend suggested using this style on  tests in another PR, and I'm a fan

Reply:
Done.